### DISCUSSION TITLE
Run validation code after each test

### DISCUSSION THREAD

**Discussion by ViktorKirichenko** [xunit/xunit#2527] [Question] [answered]
Hello,

In one class I have 3-4 tests and I need to run special validation code after each test.
I know how to run a check when all the tests in the class have finished (IDisposable). but I need to run a check after every test.
And I can't seem to find a solution.

**Comment by ViktorKirichenko** [comment]
I found how run code before every test, but how after not found :(
Do you have any idea?

**Comment by bradwilson** [comment, marked as answer]
`IDisposable` on the test class will work, because we create and destroy the test class between each (non-static) test.

If you want something reusable, you can make an attribute class which derives from [`BeforeAfterTestAttribute`](https://github.com/xunit/xunit/blob/v2/src/xunit.core/Sdk/BeforeAfterTestAttribute.cs) and decorate your test methods and/or test classes with that.
